### PR TITLE
Bump jetbrains 2024.2

### PR DESCRIPTION
**Pull request by jamesward**
## Description of changes

Bump JetBrains to 2024.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/jetbrains/bin/versions.json [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
       "update-channel": "Aqua EAP licensing:EAP",
       "url-template": "https://download.jetbrains.com/aqua/aqua-{version}.tar.gz",
-      "version": "2024.1 Public Preview",
-      "sha256": "027cac4bfab747ef97f055b63eff8a69cefc32751fcca9510c3eccaf7c36b2be",
-      "url": "https://download.jetbrains.com/aqua/aqua-241.15989.28.tar.gz",
-      "build_number": "241.15989.28"
+      "version": "2023.3 Public Preview",
+      "sha256": "7fbc5565208acd0f4428c10823778c0d2bfd192e4cfce74ee2788477dacad5e0",
+      "url": "https://download.jetbrains.com/aqua/aqua-233.14475.33.tar.gz",
+      "build_number": "233.14475.33"
```
Comment (zlepper) [inline pkgs/applications/editors/jetbrains/bin/versions.json:9]: Looks like the aqua is being downgraded for some reason. Furthermore latest build should be 242.20224.331. I think we need to switch to the Release channel now, at least looking at https://www.jetbrains.com/updates/updates.xml it looks like a "RELEASE" channel is now available and not just public preview.

**Comment by zlepper** [comment]
Result of `nixpkgs-review pr 334777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
  </ul>
</details>

Sanity tested datagrip and it works fine. Tested under X, someone might want to test under Wayland as i know this version of the IDEs contains stuff for wayland compatibility.

File: pkgs/applications/editors/jetbrains/bin/versions.json (hunk 1 above)
Comment (jamesward) [reply pkgs/applications/editors/jetbrains/bin/versions.json]: Thanks for catching that. I believe I've changed it to the `RELEASE` channel and re-run update script.

**Comment by zlepper** [comment]
@jamesward Jetbrains just released Rider 2024.2, can you update the PR? Sorry for the trouble :)

**Comment by jamesward** [comment]
@zlepper I just re-ran the update script and it didn't find any new updates.

**Comment by zlepper** [comment]
I see I was too fast, yeah, it already picked it up when you ran the scripts again earlier, apologies

**Comment by hacker1024** [comment]
This will pair well with https://github.com/NixOS/nixpkgs/pull/333421, as many of the Wayland improvements in the 2024.2 release notes are dependent on the JBR.

**Comment by hacker1024** [comment]
Result of `nixpkgs-review pr 334777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
  </ul>
</details>

**Review by hacker1024 — CHANGES_REQUESTED** [review]
The GitHub Copilot plugin does not build. I have made a PR into your branch to fix this.

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/b1x46jsgk91g4f4jkzc6gqin98r6jnm4-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
stat: cannot statx 'copilot-agent/bin/copilot-agent-linux': No such file or directory
```

File: pkgs/applications/editors/jetbrains/bin/linux.nix [hunk 2]
```diff
@@ -116,7 +116,7 @@ with stdenv; lib.makeOverridable mkDerivation (rec {
       --set-default LOCALE_ARCHIVE "${glibcLocales}/lib/locale/locale-archive" \
       --set-default ${hiName}_VM_OPTIONS ${vmoptsFile}
 
-    ln -s "$out/$pname/bin/${loName}.sh" $out/bin/$pname
+    ln -s "$out/$pname/bin/${loName}" $out/bin/$pname
```
Comment (hacker1024) [inline pkgs/applications/editors/jetbrains/bin/linux.nix:119]: This breaks for IDEs that have not yet hit 2024.2, such as RustRover.

Comment (hacker1024) [inline pkgs/applications/editors/jetbrains/bin/linux.nix:119]: This breaks for IDEs that have not yet hit 2024.2, such as RustRover.

```suggestion
    launcher="$out/$pname/bin/${loName}"
    if [ ! -e "$launcher" ]; then
      launcher="$out/$pname/bin/${loName}.sh"
    fi
    ln -s "$launcher" $out/bin/$pname
```

Comment (jamesward) [reply pkgs/applications/editors/jetbrains/bin/linux.nix]: Ah. Thanks for catching and fixing!

**Comment by jamesward** [comment]
@hacker1024 Yeah, I think we should use the JBR from https://github.com/NixOS/nixpkgs/pull/333421 because there are currently a bunch of window focus issues currently which hopefully will get better with those fixes.

**Comment by hacker1024** [comment]
> @hacker1024 Yeah, I think we should use the JBR from #333421 because there are currently a bunch of window focus issues currently which hopefully will get better with those fixes.

Feel free to merge that into this one if you think it's appropriate. Not sure what the best way to do it is

**Comment by jamesward** [comment]
I've changed this to draft and will rebase on https://github.com/NixOS/nixpkgs/pull/333421 once it is merged.

**Comment by leona-ya** [comment]
#333421 is merged now :)

**Comment by jamesward** [comment]
I've tested this with the latest JBR and it seems to be working.

**Review by leona-ya — APPROVED** [review]
LGTM, thanks!

---


Result of `nixpkgs-review pr 334777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
  </ul>
</details>

**Pull request merged**